muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/ForumController.java

```diff
@@ -224,15 +231,18 @@ public void deleteThread(ForumThread thread) {
   }
 
   public ForumThreadReply createForumThreadReply(ForumThread thread, String message, ForumThreadReply parentReply) {
-    if (thread.getLocked()) {
-      logger.severe("Tried to create a forum thread reply for locked thread");
-      return null;
-    } else {
-      ForumThreadReply reply = forumThreadReplyDAO.create(thread.getForumArea(), thread, clean(message), sessionController.getLoggedUserEntity(), parentReply);
-      forumThreadDAO.updateThreadUpdated(thread, reply.getCreated());
-      activityLogController.createActivityLog(sessionController.getLoggedUserEntity().getId(), ActivityLogType.FORUM_NEWMESSAGE);
-      return reply;
-    }
+    if (thread.getLock() != null) {
```

Comment:
Should be able to say `thread.getLock() == LockForumThread.ALL || (thread.getLock() == LockForumThread.STUDENTS` etc.